src/som/interpreter/nodes/literals/GenericArrayLiteralNode.java

```diff
@@ -0,0 +1,186 @@
+package som.interpreter.nodes.literals;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.ExplodeLoop;
+import com.oracle.truffle.api.nodes.UnexpectedResultException;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.interpreter.nodes.ExpressionNode;
+import som.vm.constants.Classes;
+import som.vmobjects.SArray;
+import som.vmobjects.SArray.SMutableArray;
+
+public class GenericArrayLiteralNode extends LiteralNode {
+  @Children protected final ExpressionNode[] expressions;
+
+  public GenericArrayLiteralNode(final List<ExpressionNode> expressions, final SourceSection source) {
+    super(source);
+    assert source != null;
+    this.expressions = expressions.toArray(new ExpressionNode[0]);
+  }
+
+  private boolean allComponentsAreBoolean(final Object[] arr) {
+    for (Object a : arr) {
+      if (!(a instanceof Boolean)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean allComponentsAreLong(final Object[] arr) {
+    for (Object a : arr) {
+      if (!(a instanceof Long)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean allComponentsAreDouble(final Object[] arr) {
+    for (Object a : arr) {
+      if (!(a instanceof Double)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean allComponentsAreString(final Object[] arr) {
+    for (Object a : arr) {
+      if (!(a instanceof String)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @ExplodeLoop
+  protected SMutableArray evaluatedExpressions(final VirtualFrame frame) {
+    Object[] arr = new Object[expressions.length];
+    for (int i = 0; i < expressions.length; i++) {
+      arr[i] = expressions[i].executeGeneric(frame);
+    }
+
+    GenericArrayLiteralNode replacedNode = this;
+
+    if (allComponentsAreBoolean(arr)) {
```

Comment:
Hm, I am not sure this is the best way of doing it. Did you see [`evaluateFirstDetermineStorageAndEvaluateRest(.)`](https://github.com/smarr/SOMns/blob/master/src/som/primitives/arrays/ArraySetAllStrategy.java#L109)? I would try to stick to existing approaches, and possibly even reuse the code.
Which might also mean that we don't need a separate class for each type of array literal.